lib/utils/validation.js

```diff
@@ -113,3 +113,33 @@ export function validateValsAgainstVocab(delimited_string, field) {
   if (update_flag) return [true, value_array.join(';')];
   else return [true, false];
 }
+
+/**
+ *
+ * @param {Array<Array<any>>} columns to validate
+ * @return {Array<boolean>}
+ *         whether each row represents a unique combination of values
+ */
+export function validateUniqueValues(values) {
+  const results = [];
+  const longestColumn = Math.max.apply(
+    null,
+    values.map((col) => col.length)
+  );
+  for (let self = 0; self < longestColumn; self += 1) {
```

Comment:
This is a loop for each row of data at least to longest index colum, say 1050 rows?

Reply:
Yes, but empty rows will have been discarded by the time it gets to this point.